core/http-auth/src/main/java/software/amazon/awssdk/http/auth/AwsV4HttpSigner.java

```diff
@@ -26,6 +29,44 @@
 @SdkPublicApi
 public interface AwsV4HttpSigner extends HttpSigner<AwsCredentialsIdentity> {
 
+    /**
+     * The datetime, in milliseconds, for the request.
+     */
+    SignerProperty<Instant> REQUEST_SIGNING_INSTANT =
+        SignerProperty.create(Instant.class, "requestSigningInstant");
+    /**
+     * The AWS region to be used for computing the signature.
+     */
+    SignerProperty<String> REGION_NAME =
+        SignerProperty.create(String.class, "regionName");
+    /**
+     * The name of the AWS service.
+     */
+    SignerProperty<String> SERVICE_SIGNING_NAME =
+        SignerProperty.create(String.class, "serviceSigningName");
+    /**
+     * The name of the header for the checksum.
+     */
+    SignerProperty<String> CHECKSUM_HEADER_NAME =
+        SignerProperty.create(String.class, "checksumHeaderName");
+    /**
+     * The Algorithm used to compute the checksum.
+     */
+    SignerProperty<Algorithm> CHECKSUM_ALGORITHM =
+        SignerProperty.create(Algorithm.class, "checksumAlgorithm");
```

Comment:
I was going to say, the "checksum" terminology doesn't seem to appear in the sigv4 public docs. They use "hash", so maybe these publicly exposed terms should match that. But the current signers seem to use "checksum" in public interfaces, so maybe staying consistent makes sense. Just putting a note here anyways, if someone else is also thinking about it.

Reply:
I think we had a debate for this in one of the SRA meetings and we settled on checksum.